src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -948,6 +948,8 @@ private static uint GetParameterRefEscape(ParameterSymbol parameter)
             {
                 { RefKind: RefKind.None } => Binder.CurrentMethodScope,
                 { EffectiveScope: DeclarationScope.RefScoped } => Binder.CurrentMethodScope,
+                { RefKind: RefKind.Out } => Binder.ReturnOnlyScope,
+                { HasUnscopedRefAttribute: true, IsThis: false } => Binder.CallingMethodScope,
```

Comment:
>IsThis: false [](http://example.com/codeflow?start=49&length=13)

Does the spec explain why the `this` parameter behaves differently? I couldn't find where that is covered.

Reply:
From Implicitly scoped parameters:

> 

Therefore `this` in a member marked `[UnscopedRef]` is _return only_.

Reply:
I agree.
From offline discussion, `this` is handled on line 950 and `[UnscopedRef]` is handled on line 952.

For `this`, the spec section copied above explains that it is implicitly-scoped and thus current-method.

For `[UnscopedRef] this` the spec says:
> When applying [UnscopedRef] to an instance method of a struct it has the impact of modifying the implicit this parameter. This means this acts as an unannotated `ref` of the same type.

That means that `[UnscopedRef] this` is return-only.